plugins/QnA/class.qna.plugin.php

```diff
@@ -1008,19 +1013,43 @@ public function discussionsController_unansweredCount_create($sender, $args) {
     }
 
     /**
+     * Print QnA meta data tag
      *
-     *
-     * @param $sender Sending controller instance.
-     * @param array $args Event arguments.
+     * @param Gdn_Controller $sender
+     * @param array $args
      */
     public function base_beforeDiscussionMeta_handler($sender, $args) {
         $discussion = $args['Discussion'];
+        if (!empty($discussion)) {
+            echo $this->getDiscussionQnATagString($discussion);
+        }
+    }
+
+    /**
+     * Print QnA meta data tag
+     *
+     * @param Gdn_Controller $sender
+     * @param array $args
+     */
+    public function discussionController_discussionInfo_handler($sender, $args) {
+        $discussion = $args['Discussion'];
+        if (!empty($discussion) && \Vanilla\FeatureFlagHelper::featureEnabled(static::FEATURE_FLAG)) {
+            echo $this->getDiscussionQnATagString($discussion);
+        }
+    }
 
+    /**
+     * Return QnA meta data tag string
+     *
+     * @param stdClass $discussion
+     */
+    private function getDiscussionQnATagString(stdClass $discussion = null): string {
```

Comment:
Can you confirm that the discussion will always be a `stdClass`? If you are fairly confident I would at least type hint to `object`.

Reply:
I've done a few tests with this coming from different controllers, it always comes as an object.

Reply:
Okay just change the type hint to give us some flexibility for the future. Checking for stdClass specifically is rarely a good idea. It stopped us from making some wanted changes in our database layer before.